packages-content-model/roosterjs-content-model-plugins/lib/autoFormat/keyboardListTrigger.ts

```diff
@@ -0,0 +1,59 @@
+import { createListItem, createListLevel } from 'roosterjs-content-model-dom';
+import { getListTypeStyle } from './utils/getListTypeStyle';
+import {
+    getOperationalBlocks,
+    isBlockGroupOfType,
+    updateListMetadata,
+} from 'roosterjs-content-model-core';
+import type { IStandaloneEditor } from 'roosterjs-content-model-types';
+
+/**
+ * @internal
+ */
+export default function keyboardListTrigger(
+    editor: IStandaloneEditor,
+    shouldSearchForBullet: boolean = true,
+    shouldSearchForNumbering: boolean = true
+) {
+    editor.formatContentModel((model, _) => {
+        const listStyleType = getListTypeStyle(
+            model,
+            shouldSearchForBullet,
+            shouldSearchForNumbering
+        );
+        if (listStyleType) {
+            const { listType, styleType, index } = listStyleType;
+
+            const paragraphOrListItems = getOperationalBlocks(model, ['ListItem'], []);
+            paragraphOrListItems.forEach(({ block, parent }, itemIndex) => {
+                if (!isBlockGroupOfType(block, 'General')) {
+                    const blockIndex = parent.blocks.indexOf(block);
```

Comment:
can we just use existing format API? e.g. `setListType`

Reply:
I tried to use it, but if we type * and space, it will create the list: 

- *

Reply:
I see. Maybe we can do some kind of refactoring. I wish we can only have one place to create the list properties.